### ISSUE TITLE
Refactor EntityTable components in DataTable components and EntityTableColumnMenu in DataTablePlusButton

### ISSUE THREAD

**Issue by bosiraphael** [twentyhq/twenty#1882]
## Scope & Context

Need to change EntityTable in DataTable for all components in front/src/modules/ui/data-table in the file names and in the components definitions.

## Technical inputs

Examples : 
- EntityTable.tsx becomes DataTable.tsx
- EntityTableBody.tsx becomes DataTableBody.tsx
...

We also need to rename EntityTableColumnMenu in DataTablePlusButton because the old name wasn't precise enough.

**Comment by b9aurav** [comment]
@bosiraphael I can work on this one, you can assign it to me.

**Comment by bosiraphael** [comment]
@b9aurav Yes, thank you ! We just have to merge another PR first. I will notify you when it's ready.

**Comment by bosiraphael** [comment]
@b9aurav The PR is merged, I assigned the issue to you! Thank you for contributing!

**Comment by b9aurav** [comment]
@bosiraphael Do we need to rename files only in `front/src/modules/ui/data-table` directory or everywhere?

There are files outside that particular directory which contains `EntityTable` in their names, like:
* `src/modules/people/components/PeopleEntityTableDataEffect.tsx`
* `src/modules/people/hooks/useSetPeopleEntityTable.tsx`
* etc..

**Comment by b9aurav** [comment]
@bosiraphael @charlesBochet If files needs to be renamed everywhere, i have my PR ready. do i submit?

**Comment by bosiraphael** [comment]
@b9aurav Yes, thank you !
